wallet/src/web.rs

```diff
@@ -254,22 +270,41 @@ async fn populatefortest(req: tide::Request<WebState>) -> Result<tide::Response,
     }
 
     // Add the faucet key, so we get a balance of the native asset.
-    wallet
-        .add_user_key(req.state().faucet_key_pair.clone(), Default::default())
-        .await
-        .unwrap();
-    wallet
-        .await_key_scan(&req.state().faucet_key_pair.address())
-        .await
-        .unwrap();
+    // Check before adding it to avoid the race condition.
+    let faucet_key_pair = req.state().faucet_key_pair.clone();
+    if !wallet.pub_keys().await.contains(&faucet_key_pair.pub_key()) {
+        wallet
+            .add_user_key(faucet_key_pair.clone(), Default::default())
+            .await
+            .unwrap();
+    }
+    let faucet_addr = faucet_key_pair.address();
+    wallet.await_key_scan(&faucet_addr).await.unwrap();
 
     // Add a wrapped asset, and give it some nonzero balance.
-    let erc20_code = Erc20Code(EthereumAddr([1; 20]));
+    let mut erc20_code = Erc20Code(EthereumAddr([0; 20]));
```

Comment:
What if we sample this from entropy? Then we're almost guaranteed to not have a collision

Reply:
Good idea! I've modified this to use entropy.